src/main/java/com/authy/api/Tokens.java

```diff
@@ -47,22 +47,23 @@ private Token tokenFromJson(int status, String content) throws AuthyException {
                 return new Token(status, content, message);
 
             } catch (JSONException e) {
-                throw new AuthyException("Invalid response from server", e);
+                throw new AuthyException("Invalid response from server", e, status);
             }
         }
 
-        Token token = new Token(status, content);
-        token.setError(errorFromJson(content));
-        return token;
+        final Error error = errorFromJson(content);
+        throw new AuthyException("Invalid token", status, error.getCode());
     }
 
     private void validateToken(String token) throws AuthyException {
         int len = token.length();
         if (!isInteger(token)) {
-            throw new AuthyException("Invalid Token. Only digits accepted.");
+            throw new AuthyException("Invalid Token. Only digits accepted.", HttpURLConnection.HTTP_UNAUTHORIZED,
+                    Error.Code.TOKEN_INVALID);
         }
         if (len < 6 || len > 10) {
```

Comment:
I thought the issue was expecting to change the validation, not just more info in the error.

Did I misunderstood that?